lib/msf/core/db_manager/client.rb

```diff
@@ -5,6 +5,7 @@ def find_or_create_client(opts)
 
   def get_client(opts)
   ::ApplicationRecord.connection_pool.with_connection {
+    opts = opts.clone() # protect the original callers array
```

Comment:
What are your thoughts on making the added comments more generic? I think some of these aren't array values, but let me know if i'm missing something :+1:

```suggestion
    opts = opts.clone() # protect the original caller's opts
```

Reply:
Totally reasonable will amend, had used comments to be clear the calls need have a purpose.

Reply:
> had used comments to be clear the calls have a purpose.

💯 

I think that's a good shout, I would never have expected this code to be mutating the original arguments :+1: